src/did/src/sign_strategy.rs

```diff
@@ -0,0 +1,334 @@
+                let address = utils::secret_key_to_address(&signer);
+                let wallet = Wallet::new_with_signer(Cow::Owned(signer), address, chain_id);
+                Ok(TxSigner::Local(LocalTxSigner::new(wallet)))
+            }
+            SigningStrategy::ManagementCanister {
+                key_id,
+                derivation_path,
+            } => Ok(TxSigner::ManagementCanister(ManagementCanisterSigner::new(
+                key_id,
+                derivation_path,
+            ))),
+        }
+    }
+}
+
+/// Transaction signer
+#[derive(Serialize, Deserialize, Clone)]
+pub enum TxSigner {
+    Local(LocalTxSigner),
+    ManagementCanister(ManagementCanisterSigner),
+}
+
+impl Default for TxSigner {
+    fn default() -> Self {
+        let signing_key = SigningKey::from_slice(&[1; 32]).expect("failed to init signing key");
+        let address = utils::secret_key_to_address(&signing_key);
+        Self::Local(LocalTxSigner {
+            wallet: Wallet::new_with_signer(Cow::Owned(signing_key), address, 0),
+        })
+    }
+}
+
+impl Storable for TxSigner {
+    fn to_bytes(&self) -> Cow<[u8]> {
+        bincode::serialize(self)
+            .expect("failed to serialize TxSigner")
+            .into()
+    }
+
+    fn from_bytes(bytes: Cow<[u8]>) -> Self {
+        bincode::deserialize(&bytes).expect("failed to deserialize TxSigner")
+    }
+}
+
+#[async_trait(?Send)]
+impl TransactionSigner for TxSigner {
+    async fn get_address(&self) -> Result<H160> {
+        match self {
+            Self::Local(signer) => signer.get_address().await,
+            Self::ManagementCanister(signer) => signer.get_address().await,
+        }
+    }
+
+    async fn sign_transaction(&self, transaction: &TypedTransaction) -> Result<Signature> {
+        match self {
+            Self::Local(signer) => signer.sign_transaction(transaction).await,
+            Self::ManagementCanister(signer) => signer.sign_transaction(transaction).await,
+        }
+    }
+}
+
+/// Local private key implementation
+#[derive(Clone)]
+pub struct LocalTxSigner {
+    wallet: Wallet<'static, SigningKey>,
+}
+
+impl LocalTxSigner {
+    fn new(wallet: Wallet<'static, SigningKey>) -> LocalTxSigner {
+        Self { wallet }
+    }
+}
+
+#[async_trait(?Send)]
+impl TransactionSigner for LocalTxSigner {
+    async fn get_address(&self) -> Result<H160> {
+        Ok(self.wallet.address().into())
+    }
+
+    async fn sign_transaction(&self, transaction: &TypedTransaction) -> Result<Signature> {
+        self.wallet
+            .sign_transaction(transaction)
+            .await
+            .map_err(|e| EvmError::from(format!("failed to sign hash: {e}")))
+            .map(Into::into)
+    }
+}
+
+/// A helper struct for serializing/deserializing `LocalTxSigner`
+#[derive(Serialize, Deserialize)]
+struct WalletSerializationData<'a> {
+    signing_key_bytes: &'a [u8],
+    address_bytes: &'a [u8],
+    chain_id: u64,
+}
+
+impl Serialize for LocalTxSigner {
```

Comment:
Why do we have all this custom serialization/deserialization code? What's the problem the serde Serialize/Deserialize derive?